### ISSUE TITLE
[Bug]: Errors encountered in 3.25+

### ISSUE THREAD

**Issue by NullVoxPopuli** [ember-cli/ember-cli#9534]
Output from `ember version --verbose && npm --version && yarn --version`:
```
❯ ember version --verbose && npm --version && yarn --version
ember-cli: 3.25.2
node: 14.15.4
v8: 8.4.371.19-node.17
uv: 1.40.0
zlib: 1.2.11
brotli: 1.0.9
ares: 1.16.1
modules: 83
nghttp2: 1.41.0
napi: 7
llhttp: 2.1.3
openssl: 1.1.1i
cldr: 37.0
icu: 67.1
tz: 2020a
unicode: 13.0
os: linux x64
6.14.10
1.22.5

```

Since 3.25 (and later), I've been getting a lot more errors with ember-cli than usual. They've been infrequent enough, where I don't know what causes them, or even if they're related. 
I'm going to be using this GH issue to log the errors I come across (as separate comments)

**Comment by NullVoxPopuli** [comment]
```
... Uhoh. Got error write EPIPE ...
Error: write EPIPE
    at WriteWrap.onWriteComplete [as oncomplete] (internal/stream_base_commons.js:94:16)
    at handleWriteReq (internal/stream_base_commons.js:53:26)
    at writeGeneric (internal/stream_base_commons.js:145:15)
    at TLSSocket.Socket._writeGeneric (net.js:785:11)
    at TLSSocket.Socket._write (net.js:797:8)
    at writeOrBuffer (internal/streams/writable.js:358:12)
    at TLSSocket.Writable.write (internal/streams/writable.js:303:10)
    at IO.ondata (internal/streams/legacy.js:19:31)
    at IO.emit (events.js:315:20)
    at IO.EventEmitter.emit (domain.js:467:12)
    at Hybi._write (/home/me/my-project/node_modules/websocket-driver/lib/websocket/driver/base.js:158:25)
    at Hybi._sendFrame (/home/me/my-project/node_modules/websocket-driver/lib/websocket/driver/hybi.js:270:10)
    at Hybi.onMessageReady (/home/me/my-project/node_modules/websocket-driver/lib/websocket/driver/hybi.js:225:12)
    at Hybi.frame (/home/me/my-project/node_modules/websocket-driver/lib/websocket/driver/hybi.js:234:22)
    at Hybi.<anonymous> (/home/me/my-project/node_modules/websocket-driver/lib/websocket/driver/hybi.js:307:32)
    at /home/me/my-project/node_modules/websocket-extensions/lib/pipeline/index.js:26:51
    at Pledge.then (/home/me/my-project/node_modules/websocket-extensions/lib/pipeline/pledge.js:27:23)
    at Pipeline.close (/home/me/my-project/node_modules/websocket-extensions/lib/pipeline/index.js:26:24)
    at Extensions.close (/home/me/my-project/node_modules/websocket-extensions/lib/websocket_extensions.js:142:20)
    at Hybi._shutdown (/home/me/my-project/node_modules/websocket-driver/lib/websocket/driver/hybi.js:306:22)
    at Hybi._emitFrame (/home/me/my-project/node_modules/websocket-driver/lib/websocket/driver/hybi.js:425:12)
    at Hybi.parse (/home/me/my-project/node_modules/websocket-driver/lib/websocket/driver/hybi.js:137:18)
cleaning up...

```
I wasn't looking at the computer when this happened. I walked away for a while and the computer went to sleep

**Comment by NullVoxPopuli** [comment]
New error, was just editing a test file when this happened:

<details><summary>console output</summary>

```
internal/process/promises.js:213
        triggerUncaughtException(err, true /* fromPromise */);
        ^

ModuleBuildError: Module build failed (from ../../../../home/me/Development/NullVoxPopuli/limber/node_modules/thread-loader/dist/cjs.js):
Thread Loader (Worker 0)
/tmp/embroider/19ab55/frontend/tests/application/-page.ts: Unexpected token, expected "," (34:58)

  32 |
  33 |         let { origin, pathname } = this.element.href;
> 34 |         console.log(currentURL(), { origin, pathname, this.element });
     |                                                           ^
  35 |
  36 |         return currentURL().startsWith(this.element.href);
  37 |       }

    at Object._raise (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:810:17)
    at Object.raiseWithData (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:803:17)
    at Object.raise (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:764:17)
    at Object.unexpected (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:9965:16)
    at Object.expect (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:9939:28)
    at Object.parseObjectLike (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:11725:14)
    at Object.parseExprAtom (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:11284:23)
    at Object.parseExprSubscripts (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:10938:23)
    at Object.parseUpdate (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:10918:21)
    at Object.parseMaybeUnary (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:10896:23)
    at processResult (/home/me/Development/NullVoxPopuli/limber/node_modules/webpack/lib/NormalModule.js:690:19)
    at /home/me/Development/NullVoxPopuli/limber/node_modules/webpack/lib/NormalModule.js:792:5
    at /home/me/Development/NullVoxPopuli/limber/node_modules/loader-runner/lib/LoaderRunner.js:399:11
    at /home/me/Development/NullVoxPopuli/limber/node_modules/loader-runner/lib/LoaderRunner.js:195:20
    at context.callback (/home/me/Development/NullVoxPopuli/limber/node_modules/loader-runner/lib/LoaderRunner.js:124:13)
    at Object.callback (/home/me/Development/NullVoxPopuli/limber/node_modules/thread-loader/dist/index.js:52:7)
    at done (/home/me/Development/NullVoxPopuli/limber/node_modules/neo-async/async.js:8069:18)
    at callback (/home/me/Development/NullVoxPopuli/limber/node_modules/thread-loader/dist/WorkerPool.js:185:19)
    at /home/me/Development/NullVoxPopuli/limber/node_modules/thread-loader/dist/WorkerPool.js:219:15
    at mapSeries (/home/me/Development/NullVoxPopuli/limber/node_modules/neo-async/async.js:3625:14) {
  isBuilderError: true,
  isSilent: undefined,
  isCancellation: undefined,
  broccoliPayload: {
    originalError: <ref *1> ModuleBuildError: Module build failed (from ../../../../home/me/Development/NullVoxPopuli/limber/node_modules/thread-loader/dist/cjs.js):
    Thread Loader (Worker 0)
    /tmp/embroider/19ab55/frontend/tests/application/-page.ts: Unexpected token, expected "," (34:58)
    
      32 |
      33 |         let { origin, pathname } = this.element.href;
    > 34 |         console.log(currentURL(), { origin, pathname, this.element });
         |                                                           ^
      35 |
      36 |         return currentURL().startsWith(this.element.href);
      37 |       }
    
        at Object._raise (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:810:17)
        at Object.raiseWithData (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:803:17)
        at Object.raise (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:764:17)
        at Object.unexpected (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:9965:16)
        at Object.expect (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:9939:28)
        at Object.parseObjectLike (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:11725:14)
        at Object.parseExprAtom (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:11284:23)
        at Object.parseExprSubscripts (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:10938:23)
        at Object.parseUpdate (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:10918:21)
        at Object.parseMaybeUnary (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:10896:23)
        at processResult (/home/me/Development/NullVoxPopuli/limber/node_modules/webpack/lib/NormalModule.js:690:19)
        at /home/me/Development/NullVoxPopuli/limber/node_modules/webpack/lib/NormalModule.js:792:5
        at /home/me/Development/NullVoxPopuli/limber/node_modules/loader-runner/lib/LoaderRunner.js:399:11
        at /home/me/Development/NullVoxPopuli/limber/node_modules/loader-runner/lib/LoaderRunner.js:195:20
        at context.callback (/home/me/Development/NullVoxPopuli/limber/node_modules/loader-runner/lib/LoaderRunner.js:124:13)
        at Object.callback (/home/me/Development/NullVoxPopuli/limber/node_modules/thread-loader/dist/index.js:52:7)
        at done (/home/me/Development/NullVoxPopuli/limber/node_modules/neo-async/async.js:8069:18)
        at callback (/home/me/Development/NullVoxPopuli/limber/node_modules/thread-loader/dist/WorkerPool.js:185:19)
        at /home/me/Development/NullVoxPopuli/limber/node_modules/thread-loader/dist/WorkerPool.js:219:15
        at mapSeries (/home/me/Development/NullVoxPopuli/limber/node_modules/neo-async/async.js:3625:14) {
      details: undefined,
      module: NormalModule {
        dependencies: [],
        blocks: [],
        type: 'javascript/auto',
        context: '/tmp/embroider/19ab55/frontend/tests/application',
        layer: null,
        needId: true,
        debugId: 5955,
        resolveOptions: {},
        factoryMeta: undefined,
        useSourceMap: false,
        useSimpleSourceMap: false,
        _warnings: undefined,
        _errors: [ [Circular *1] ],
        buildMeta: { exportsType: 'namespace' },
        buildInfo: {
          cacheable: true,
          parsed: true,
          fileDependencies: LazySet {
            _set: Set(0) {},
            _toMerge: Set(0) {},
            _toDeepMerge: [Array],
            _needMerge: true,
            _deopt: false
          },
          contextDependencies: LazySet {
            _set: Set(0) {},
            _toMerge: Set(0) {},
            _toDeepMerge: [Array],
            _needMerge: true,
            _deopt: false
          },
          missingDependencies: LazySet {
            _set: Set(0) {},
            _toMerge: Set(0) {},
            _toDeepMerge: [Array],
            _needMerge: true,
            _deopt: false
          },
          buildDependencies: LazySet {
            _set: [Set],
            _toMerge: Set(0) {},
            _toDeepMerge: [],
            _needMerge: false,
            _deopt: false
          },
          valueDependencies: undefined,
          hash: 'dc436ce4731bc9f1eaee512bb513d126',
          assets: undefined,
          assetsInfo: undefined
        },
        presentationalDependencies: [],
        request: '/home/me/Development/NullVoxPopuli/limber/node_modules/thread-loader/dist/cjs.js??ruleSet[1].rules[1].use[0]!/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/lib/index.js??ruleSet[1].rules[1].use[1]!/tmp/embroider/19ab55/frontend/tests/application/-page.ts',
        userRequest: '/tmp/embroider/19ab55/frontend/tests/application/-page.ts',
        rawRequest: './-page',
        binary: false,
        parser: JavascriptParser {
          hooks: {
            evaluateTypeof: [HookMap],
            evaluate: [HookMap],
            evaluateIdentifier: [HookMap],
            evaluateDefinedIdentifier: [HookMap],
            evaluateCallExpressionMember: [HookMap],
            isPure: [HookMap],
            preStatement: [Hook],
            blockPreStatement: [Hook],
            statement: [Hook],
            statementIf: [Hook],
            classExtendsExpression: [Hook],
            classBodyElement: [Hook],
            classBodyValue: [Hook],
            label: [HookMap],
            import: [Hook],
            importSpecifier: [Hook],
            export: [Hook],
            exportImport: [Hook],
            exportDeclaration: [Hook],
            exportExpression: [Hook],
            exportSpecifier: [Hook],
            exportImportSpecifier: [Hook],
            preDeclarator: [Hook],
            declarator: [Hook],
            varDeclaration: [HookMap],
            varDeclarationLet: [HookMap],
            varDeclarationConst: [HookMap],
            varDeclarationVar: [HookMap],
            pattern: [HookMap],
            canRename: [HookMap],
            rename: [HookMap],
            assign: [HookMap],
            assignMemberChain: [HookMap],
            typeof: [HookMap],
            importCall: [Hook],
            topLevelAwait: [Hook],
            call: [HookMap],
            callMemberChain: [HookMap],
            memberChainOfCallMemberChain: [HookMap],
            callMemberChainOfCallMemberChain: [HookMap],
            optionalChaining: [Hook],
            new: [HookMap],
            expression: [HookMap],
            expressionMemberChain: [HookMap],
            unhandledExpressionMemberChain: [HookMap],
            expressionConditionalOperator: [Hook],
            expressionLogicalOperator: [Hook],
            program: [Hook],
            finish: [Hook]
          },
          sourceType: 'auto',
          scope: undefined,
          state: undefined,
          comments: undefined,
          semicolons: undefined,
          statementPath: undefined,
          prevStatement: undefined,
          currentTagData: undefined
        },
        parserOptions: undefined,
        generator: JavascriptGenerator {},
        generatorOptions: undefined,
        resource: '/tmp/embroider/19ab55/frontend/tests/application/-page.ts',
        matchResource: undefined,
        loaders: [
          {
            loader: '/home/me/Development/NullVoxPopuli/limber/node_modules/thread-loader/dist/cjs.js',
            options: [Object],
            ident: 'ruleSet[1].rules[1].use[0]'
          },
          {
            loader: '/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/lib/index.js',
            options: [Object],
            ident: 'ruleSet[1].rules[1].use[1]'
          }
        ],
        error: [Circular *1],
        _source: null,
        _sourceSizes: Map(2) { 'javascript' => 39, undefined => 39 },
        _lastSuccessfulBuildMeta: { exportsType: 'namespace' },
        _forceBuild: false,
        _isEvaluatingSideEffects: false,
        _addedSideEffectsBailout: undefined,
        _ast: null
      },
      loc: undefined,
      hideStack: undefined,
      chunk: undefined,
      file: 'tests/application/-page.ts',
      error: Thread Loader (Worker 0)
      /tmp/embroider/19ab55/frontend/tests/application/-page.ts: Unexpected token, expected "," (34:58)
      
        32 |
        33 |         let { origin, pathname } = this.element.href;
      > 34 |         console.log(currentURL(), { origin, pathname, this.element });
           |                                                           ^
        35 |
        36 |         return currentURL().startsWith(this.element.href);
        37 |       }
      
          at Object._raise (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:810:17)
          at Object.raiseWithData (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:803:17)
          at Object.raise (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:764:17)
          at Object.unexpected (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:9965:16)
          at Object.expect (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:9939:28)
          at Object.parseObjectLike (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:11725:14)
          at Object.parseExprAtom (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:11284:23)
          at Object.parseExprSubscripts (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:10938:23)
          at Object.parseUpdate (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:10918:21)
          at Object.parseMaybeUnary (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:10896:23) {
        name: undefined
      },
      line: null
    },
    originalMessage: 'Module build failed (from ../../../../home/me/Development/NullVoxPopuli/limber/node_modules/thread-loader/dist/cjs.js):\n' +
      'Thread Loader (Worker 0)\n' +
      '$TMPDIR/embroider/19ab55/frontend/tests/application/-page.ts/-page.ts: Unexpected token, expected "," (34:58)\n' +
      '\n' +
      '  32 |\n' +
      '  33 |         let { origin, pathname } = this.element.href;\n' +
      '> 34 |         console.log(currentURL(), { origin, pathname, this.element });\n' +
      '     |                                                           ^\n' +
      '  35 |\n' +
      '  36 |         return currentURL().startsWith(this.element.href);\n' +
      '  37 |       }\n' +
      '\n' +
      '    at Object._raise (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:810:17)\n' +
      '    at Object.raiseWithData (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:803:17)\n' +
      '    at Object.raise (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:764:17)\n' +
      '    at Object.unexpected (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:9965:16)\n' +
      '    at Object.expect (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:9939:28)\n' +
      '    at Object.parseObjectLike (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:11725:14)\n' +
      '    at Object.parseExprAtom (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:11284:23)\n' +
      '    at Object.parseExprSubscripts (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:10938:23)\n' +
      '    at Object.parseUpdate (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:10918:21)\n' +
      '    at Object.parseMaybeUnary (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:10896:23)',
    nodeId: 208,
    nodeLabel: 'PackagerRunner (@embroider/webpack)',
    nodeName: 'PackagerRunner',
    nodeAnnotation: '@embroider/webpack',
    instantiationStack: '    at new Plugin (/home/me/Development/NullVoxPopuli/limber/node_modules/@embroider/core/node_modules/broccoli-plugin/dist/index.js:47:36)\n' +
      '    at new PackagerRunner (/home/me/Development/NullVoxPopuli/limber/node_modules/@embroider/core/src/to-broccoli-plugin.js:10:13)\n' +
      '    at Object.defaultPipeline (/home/me/Development/NullVoxPopuli/limber/node_modules/@embroider/compat/src/default-pipeline.js:43:12)\n' +
      '    at module.exports (/home/me/Development/NullVoxPopuli/limber/frontend/ember-cli-build.js:48:39)\n' +
      '    at Builder.readBuildFile (/home/me/Development/NullVoxPopuli/limber/node_modules/ember-cli/lib/models/builder.js:49:14)\n' +
      '    at Builder.setupBroccoliBuilder (/home/me/Development/NullVoxPopuli/limber/node_modules/ember-cli/lib/models/builder.js:63:22)\n' +
      '    at new Builder (/home/me/Development/NullVoxPopuli/limber/node_modules/ember-cli/lib/models/builder.js:29:10)\n' +
      '    at ServeTask.run (/home/me/Development/NullVoxPopuli/limber/node_modules/ember-cli/lib/tasks/serve.js:49:7)\n' +
      '    at /home/me/Development/NullVoxPopuli/limber/node_modules/ember-cli/lib/models/command.js:238:24\n' +
      '    at processTicksAndRejections (internal/process/task_queues.js:95:5)',
    location: {
      file: 'tests/application/-page.ts',
      treeDir: undefined,
      line: null,
      column: undefined
    },
    error: {
      message: 'Module build failed (from ../../../../home/me/Development/NullVoxPopuli/limber/node_modules/thread-loader/dist/cjs.js):\n' +
        'Thread Loader (Worker 0)\n' +
        '$TMPDIR/embroider/19ab55/frontend/tests/application/-page.ts/-page.ts: Unexpected token, expected "," (34:58)\n' +
        '\n' +
        '  32 |\n' +
        '  33 |         let { origin, pathname } = this.element.href;\n' +
        '> 34 |         console.log(currentURL(), { origin, pathname, this.element });\n' +
        '     |                                                           ^\n' +
        '  35 |\n' +
        '  36 |         return currentURL().startsWith(this.element.href);\n' +
        '  37 |       }\n' +
        '\n' +
        '    at Object._raise (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:810:17)\n' +
        '    at Object.raiseWithData (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:803:17)\n' +
        '    at Object.raise (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:764:17)\n' +
        '    at Object.unexpected (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:9965:16)\n' +
        '    at Object.expect (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:9939:28)\n' +
        '    at Object.parseObjectLike (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:11725:14)\n' +
        '    at Object.parseExprAtom (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:11284:23)\n' +
        '    at Object.parseExprSubscripts (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:10938:23)\n' +
        '    at Object.parseUpdate (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:10918:21)\n' +
        '    at Object.parseMaybeUnary (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:10896:23)',
      stack: 'ModuleBuildError: Module build failed (from ../../../../home/me/Development/NullVoxPopuli/limber/node_modules/thread-loader/dist/cjs.js):\n' +
        'Thread Loader (Worker 0)\n' +
        '/tmp/embroider/19ab55/frontend/tests/application/-page.ts: Unexpected token, expected "," (34:58)\n' +
        '\n' +
        '  32 |\n' +
        '  33 |         let { origin, pathname } = this.element.href;\n' +
        '> 34 |         console.log(currentURL(), { origin, pathname, this.element });\n' +
        '     |                                                           ^\n' +
        '  35 |\n' +
        '  36 |         return currentURL().startsWith(this.element.href);\n' +
        '  37 |       }\n' +
        '\n' +
        '    at Object._raise (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:810:17)\n' +
        '    at Object.raiseWithData (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:803:17)\n' +
        '    at Object.raise (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:764:17)\n' +
        '    at Object.unexpected (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:9965:16)\n' +
        '    at Object.expect (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:9939:28)\n' +
        '    at Object.parseObjectLike (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:11725:14)\n' +
        '    at Object.parseExprAtom (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:11284:23)\n' +
        '    at Object.parseExprSubscripts (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:10938:23)\n' +
        '    at Object.parseUpdate (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:10918:21)\n' +
        '    at Object.parseMaybeUnary (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:10896:23)\n' +
        '    at processResult (/home/me/Development/NullVoxPopuli/limber/node_modules/webpack/lib/NormalModule.js:690:19)\n' +
        '    at /home/me/Development/NullVoxPopuli/limber/node_modules/webpack/lib/NormalModule.js:792:5\n' +
        '    at /home/me/Development/NullVoxPopuli/limber/node_modules/loader-runner/lib/LoaderRunner.js:399:11\n' +
        '    at /home/me/Development/NullVoxPopuli/limber/node_modules/loader-runner/lib/LoaderRunner.js:195:20\n' +
        '    at context.callback (/home/me/Development/NullVoxPopuli/limber/node_modules/loader-runner/lib/LoaderRunner.js:124:13)\n' +
        '    at Object.callback (/home/me/Development/NullVoxPopuli/limber/node_modules/thread-loader/dist/index.js:52:7)\n' +
        '    at done (/home/me/Development/NullVoxPopuli/limber/node_modules/neo-async/async.js:8069:18)\n' +
        '    at callback (/home/me/Development/NullVoxPopuli/limber/node_modules/thread-loader/dist/WorkerPool.js:185:19)\n' +
        '    at /home/me/Development/NullVoxPopuli/limber/node_modules/thread-loader/dist/WorkerPool.js:219:15\n' +
        '    at mapSeries (/home/me/Development/NullVoxPopuli/limber/node_modules/neo-async/async.js:3625:14)',
      errorType: 'Build Error',
      codeFrame: 'Module build failed (from ../../../../home/me/Development/NullVoxPopuli/limber/node_modules/thread-loader/dist/cjs.js):\n' +
        'Thread Loader (Worker 0)\n' +
        '$TMPDIR/embroider/19ab55/frontend/tests/application/-page.ts/-page.ts: Unexpected token, expected "," (34:58)\n' +
        '\n' +
        '  32 |\n' +
        '  33 |         let { origin, pathname } = this.element.href;\n' +
        '> 34 |         console.log(currentURL(), { origin, pathname, this.element });\n' +
        '     |                                                           ^\n' +
        '  35 |\n' +
        '  36 |         return currentURL().startsWith(this.element.href);\n' +
        '  37 |       }\n' +
        '\n' +
        '    at Object._raise (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:810:17)\n' +
        '    at Object.raiseWithData (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:803:17)\n' +
        '    at Object.raise (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:764:17)\n' +
        '    at Object.unexpected (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:9965:16)\n' +
        '    at Object.expect (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:9939:28)\n' +
        '    at Object.parseObjectLike (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:11725:14)\n' +
        '    at Object.parseExprAtom (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:11284:23)\n' +
        '    at Object.parseExprSubscripts (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:10938:23)\n' +
        '    at Object.parseUpdate (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:10918:21)\n' +
        '    at Object.parseMaybeUnary (/home/me/Development/NullVoxPopuli/limber/node_modules/babel-loader/node_modules/@babel/parser/lib/index.js:10896:23)',
      location: {
        file: 'tests/application/-page.ts',
        treeDir: undefined,
        line: null,
        column: undefined
      }
    },
    broccoliNode: {
      nodeName: 'PackagerRunner',
      nodeAnnotation: '@embroider/webpack',
      instantiationStack: '    at new Plugin (/home/me/Development/NullVoxPopuli/limber/node_modules/@embroider/core/node_modules/broccoli-plugin/dist/index.js:47:36)\n' +
        '    at new PackagerRunner (/home/me/Development/NullVoxPopuli/limber/node_modules/@embroider/core/src/to-broccoli-plugin.js:10:13)\n' +
        '    at Object.defaultPipeline (/home/me/Development/NullVoxPopuli/limber/node_modules/@embroider/compat/src/default-pipeline.js:43:12)\n' +
        '    at module.exports (/home/me/Development/NullVoxPopuli/limber/frontend/ember-cli-build.js:48:39)\n' +
        '    at Builder.readBuildFile (/home/me/Development/NullVoxPopuli/limber/node_modules/ember-cli/lib/models/builder.js:49:14)\n' +
        '    at Builder.setupBroccoliBuilder (/home/me/Development/NullVoxPopuli/limber/node_modules/ember-cli/lib/models/builder.js:63:22)\n' +
        '    at new Builder (/home/me/Development/NullVoxPopuli/limber/node_modules/ember-cli/lib/models/builder.js:29:10)\n' +
        '    at ServeTask.run (/home/me/Development/NullVoxPopuli/limber/node_modules/ember-cli/lib/tasks/serve.js:49:7)\n' +
        '    at /home/me/Development/NullVoxPopuli/limber/node_modules/ember-cli/lib/models/command.js:238:24\n' +
        '    at processTicksAndRejections (internal/process/task_queues.js:95:5)'
    },
    versions: { 'broccoli-builder': '3.5.2', node: 'v14.17.0' }
  }
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed.
Exit code: 1
Command: /home/me/.volta/tools/image/node/14.17.0/bin/node
Arguments: /home/me/.volta/tools/image/yarn/1.22.10/lib/cli.js start
Directory: /home/me/Development/NullVoxPopuli/limber/frontend
Output:

info Visit https://yarnpkg.com/en/docs/cli/workspace for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

</details>


<details><summary>ENV Summary in the log with the same info as above</summary>

```
ENV Summary:

  TIME: Sat May 15 2021 11:52:36 GMT-0400 (Eastern Daylight Time)
  TITLE: ember
  ARGV:
  - /home/me/.volta/tools/image/node/14.17.0/bin/node
  - /home/me/Development/NullVoxPopuli/limber/frontend/node_modules/.bin/ember
  - serve
  EXEC_PATH: /home/me/.volta/tools/image/node/14.17.0/bin/node
  TMPDIR: /tmp
  SHELL: /bin/bash
  PATH:
  - /tmp/yarn--1621091920506-0.5801122067254441
  - /home/me/Development/NullVoxPopuli/limber/frontend/node_modules/.bin
  - /home/me/.config/yarn/link/node_modules/.bin
  - /home/me/Development/NullVoxPopuli/limber/node_modules/.bin
  - /home/me/.volta/tools/image/node/14.17.0/libexec/lib/node_modules/npm/bin/node-gyp-bin
  - /home/me/.volta/tools/image/node/14.17.0/lib/node_modules/npm/bin/node-gyp-bin
  - /home/me/.volta/tools/image/node/14.17.0/bin/node_modules/npm/bin/node-gyp-bin
  - /tmp/yarn--1621091920114-0.020758875406385213
  - /home/me/Development/NullVoxPopuli/limber/node_modules/.bin
  - /home/me/.config/yarn/link/node_modules/.bin
  - /home/me/Development/NullVoxPopuli/limber/node_modules/.bin
  - /home/me/.yarn/bin
  - /home/me/.volta/tools/image/node/14.17.0/libexec/lib/node_modules/npm/bin/node-gyp-bin
  - /home/me/.volta/tools/image/node/14.17.0/lib/node_modules/npm/bin/node-gyp-bin
  - /home/me/.volta/tools/image/node/14.17.0/bin/node_modules/npm/bin/node-gyp-bin
  - /home/me/.volta/tools/image/npm/7.11.1/bin
  - /home/me/.volta/tools/image/yarn/1.22.10/bin
  - /home/me/.volta/tools/image/node/14.17.0/bin
  - /home/me/.volta/bin
  - /home/me/.pythons/Python-3.6.3/bin
  - /home/me/.cargo/bin
  - /home/me/Applications
  - /home/me/apps/phantomjs/bin
  - /home/me/scripts/system-utils
  - /home/me/scripts/git
  - /home/me/scripts/rails
  - /home/me/scripts
  - /home/me/.volta/bin
  - /usr/local/sbin
  - /usr/local/bin
  - /usr/sbin
  - /usr/bin
  - /sbin
  - /bin
  - /usr/games
  - /usr/local/games
  - /snap/bin
  - /home/me/.dotnet/tools
  - /home/me/.dotnet/tools
  - /home/me/.fzf/bin
  PLATFORM: linux x64
  FREEMEM: 11064426496
  TOTALMEM: 17388204032
  UPTIME: 4989.43
  LOADAVG: 0.39,0.34,0.36
  CPUS:
  - Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz - 3192
  - Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz - 3192
  - Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz - 3192
  - Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz - 3192
  - Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz - 3192
  - Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz - 3192
  ENDIANNESS: LE
  VERSIONS:
  - ares: 1.17.1
  - brotli: 1.0.9
  - cldr: 38.1
  - icu: 68.2
  - llhttp: 2.1.3
  - modules: 83
  - napi: 8
  - nghttp2: 1.42.0
  - node: 14.17.0
  - openssl: 1.1.1k
  - tz: 2020d
  - unicode: 13.0
  - uv: 1.41.0
  - v8: 8.4.371.23-node.63
  - zlib: 1.2.11

```

</details>

**Comment by NullVoxPopuli** [comment]
This one was reported by a co-worker,

new error, different app/repo -- not using embroider, also monorepo on Mac OS.


This error occurred after Ember froze. ctrl+c was hit, so the cancellation request makes sense, but there is no information about what lead up to the freeze. 

After both errors (in separate terminals (Apple's default)), the terminal no longer rendered input.

<details><summary>cancellation error</summary>

```
=================================================================================
ENV Summary:
  TIME: Mon May 17 2021 10:43:19 GMT-0500 (Central Daylight Time)
  TITLE: ember
  ARGV:
  - /Users/me/.volta/tools/image/node/14.15.4/bin/node
  - /Users/me/Projects/my-monorepo/my-app-workspace/node_modules/.bin/ember
  - s
  EXEC_PATH: /Users/me/.volta/tools/image/node/14.15.4/bin/node
  TMPDIR: /var/folders/jc/pckth1zj4pn8_8_2__qydnc80000gn/T
  SHELL: /bin/zsh
  PATH:
  - /Users/me/.volta/tools/image/yarn/1.22.5/bin
  - /Users/me/.volta/tools/image/node/14.15.4/bin
  - /Users/me/.yarn/bin
  - /usr/local/bin
  - /usr/bin
  - /bin
  - /usr/sbin
  - /sbin
  - /Library/Apple/usr/bin
  - /Users/me/.volta/bin
  - /usr/local/sbin
  - /Users/me/.yarn/bin
  PLATFORM: darwin x64
  FREEMEM: 982380544
  TOTALMEM: 17179869184
  UPTIME: 498566
  LOADAVG: 3.68798828125,3.9248046875,3.35107421875
  CPUS:
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  ENDIANNESS: LE
  VERSIONS:
  - ares: 1.16.1
  - brotli: 1.0.9
  - cldr: 37.0
  - icu: 67.1
  - llhttp: 2.1.3
  - modules: 83
  - napi: 7
  - nghttp2: 1.41.0
  - node: 14.15.4
  - openssl: 1.1.1i
  - tz: 2020a
  - unicode: 13.0
  - uv: 1.40.0
  - v8: 8.4.371.19-node.17
  - zlib: 1.2.11
ERROR Summary:
  - broccoliBuilderErrorStack: [undefined]
  - code: [undefined]
  - codeFrame: [undefined]
  - errorMessage: Build Canceled
  - errorType: [undefined]
  - location:
    - column: [undefined]
    - file: [undefined]
    - line: [undefined]
  - message: Build Canceled
  - name: Error
  - nodeAnnotation: [undefined]
  - nodeName: [undefined]
  - originalErrorMessage: [undefined]
  - stack: Error: Build Canceled
    at CancelationRequest.throwIfRequested (/Users/me/Projects/my-monorepo/node_modules/broccoli/dist/cancelation-request.js:17:19)
    at /Users/me/Projects/my-monorepo/node_modules/broccoli/dist/builder.js:182:42
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
=================================================================================
```

</details>



<details><summary>EMPTY_MERGE_TREE error log after trying to restart the dev server</summary>

```

=================================================================================
ENV Summary:
  TIME: Mon May 17 2021 12:44:40 GMT-0500 (Central Daylight Time)
  TITLE: ember
  ARGV:
  - /Users/me/.volta/tools/image/node/14.15.4/bin/node
  - /Users/me/Projects/my-monorepo/my-addon-workspace/node_modules/.bin/ember
  - test
  - --server
  - --no-launch
  EXEC_PATH: /Users/me/.volta/tools/image/node/14.15.4/bin/node
  TMPDIR: /var/folders/jc/pckth1zj4pn8_8_2__qydnc80000gn/T
  SHELL: /bin/zsh
  PATH:
  - /Users/me/.volta/tools/image/yarn/1.22.5/bin
  - /Users/me/.volta/tools/image/node/14.15.4/bin
  - /Users/me/.yarn/bin
  - /usr/local/bin
  - /usr/bin
  - /bin
  - /usr/sbin
  - /sbin
  - /Library/Apple/usr/bin
  - /Users/me/.volta/bin
  - /usr/local/sbin
  - /Users/me/.yarn/bin
  PLATFORM: darwin x64
  FREEMEM: 2492194816
  TOTALMEM: 17179869184
  UPTIME: 505847
  LOADAVG: 2.28857421875,2.94580078125,3.36181640625
  CPUS:
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz - 2600
  ENDIANNESS: LE
  VERSIONS:
  - ares: 1.16.1
  - brotli: 1.0.9
  - cldr: 37.0
  - icu: 67.1
  - llhttp: 2.1.3
  - modules: 83
  - napi: 7
  - nghttp2: 1.41.0
  - node: 14.15.4
  - openssl: 1.1.1i
  - tz: 2020a
  - unicode: 13.0
  - uv: 1.40.0
  - v8: 8.4.371.19-node.17
  - zlib: 1.2.11
ERROR Summary:
  - broccoliBuilderErrorStack: Error: [BroccoliMergeTrees: EMPTY_MERGE_TREE] error while merging the following:
ENOENT: no such file or directory, mkdir '/var/folders/jc/pckth1zj4pn8_8_2__qydnc80000gn/T/broccoli-98078d7QG7SJvr6mU/out-000-broccoli_merge_trees_empty_merge_tree'
    at Object.mkdirSync (fs.js:987:3)
    at update (/Users/me/Projects/my-monorepo/node_modules/fs-updater/index.js:186:10)
    at FSUpdater._update (/Users/me/Projects/my-monorepo/node_modules/fs-updater/index.js:116:7)
    at FSUpdater._update (/Users/me/Projects/my-monorepo/node_modules/fs-updater/index.js:121:12)
    at FSUpdater.update (/Users/me/Projects/my-monorepo/node_modules/fs-updater/index.js:105:10)
    at MergeTrees.merge (/Users/me/Projects/my-monorepo/node_modules/broccoli-merge-trees/node_modules/merge-trees/index.js:31:18)
    at BroccoliMergeTrees.build (/Users/me/Projects/my-monorepo/node_modules/broccoli-merge-trees/index.js:36:21)
    at TransformNodeWrapper.build (/Users/me/Projects/my-monorepo/node_modules/broccoli/dist/wrappers/transform-node.js:71:39)
    at /Users/me/Projects/my-monorepo/node_modules/broccoli/dist/builder.js:185:30
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Builder.build (/Users/me/Projects/my-monorepo/node_modules/broccoli/dist/builder.js:204:13)
    at async Builder.build (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/models/builder.js:197:11)
  - code: [undefined]
  - codeFrame: [BroccoliMergeTrees: EMPTY_MERGE_TREE] error while merging the following:
ENOENT: no such file or directory, mkdir '/var/folders/jc/pckth1zj4pn8_8_2__qydnc80000gn/T/broccoli-98078d7QG7SJvr6mU/out-000-broccoli_merge_trees_empty_merge_tree'
  - errorMessage: [BroccoliMergeTrees: EMPTY_MERGE_TREE] error while merging the following:
ENOENT: no such file or directory, mkdir '/var/folders/jc/pckth1zj4pn8_8_2__qydnc80000gn/T/broccoli-98078d7QG7SJvr6mU/out-000-broccoli_merge_trees_empty_merge_tree'
        at BroccoliMergeTrees (EMPTY_MERGE_TREE)
-~- created here: -~-
    at BroccoliMergeTrees.Plugin (/Users/me/Projects/my-monorepo/node_modules/broccoli-plugin/index.js:7:31)
    at new BroccoliMergeTrees (/Users/me/Projects/my-monorepo/node_modules/broccoli-merge-trees/index.js:16:10)
    at Function.BroccoliMergeTrees [as _upstreamMergeTrees] (/Users/me/Projects/my-monorepo/node_modules/broccoli-merge-trees/index.js:10:53)
    at getEmptyTree (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/broccoli/merge-trees.js:17:37)
    at mergeTrees (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/broccoli/merge-trees.js:82:14)
    at Class.treeFor (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/models/addon.js:638:30)
    at /Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/models/addon.js:497:26
    at Array.reduce (<anonymous>)
    at Class.eachAddonInvoke (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/models/addon.js:494:24)
    at Class.treeFor (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/models/addon.js:627:22)
    at /Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:697:26
    at Array.reduce (<anonymous>)
    at EmberAddon._addonTreesFor (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:695:32)
    at EmberAddon.addonTreesFor (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:719:17)
    at EmberAddon.getAddonTemplates (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:1067:27)
    at EmberAddon.toArray (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:1537:12)
    at EmberAddon.toTree (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:1644:36)
    at EmberAddon.appInstance.toTree (/Users/me/Projects/my-monorepo/node_modules/@embroider/macros/src/ember-addon-main.js:48:35)
    at module.exports (/Users/me/Projects/my-monorepo/my-addon-workspace/ember-cli-build.js:12:14)
    at Builder.readBuildFile (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/models/builder.js:49:14)
    at Builder.setupBroccoliBuilder (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/models/builder.js:63:22)
    at new Builder (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/models/builder.js:29:10)
-~- (end) -~-
  - errorType: Build Error
  - location:
    - column: [undefined]
    - file: [undefined]
    - line: [undefined]
    - treeDir: [undefined]
  - message: [BroccoliMergeTrees: EMPTY_MERGE_TREE] error while merging the following:
ENOENT: no such file or directory, mkdir '/var/folders/jc/pckth1zj4pn8_8_2__qydnc80000gn/T/broccoli-98078d7QG7SJvr6mU/out-000-broccoli_merge_trees_empty_merge_tree'
        at BroccoliMergeTrees (EMPTY_MERGE_TREE)
-~- created here: -~-
    at BroccoliMergeTrees.Plugin (/Users/me/Projects/my-monorepo/node_modules/broccoli-plugin/index.js:7:31)
    at new BroccoliMergeTrees (/Users/me/Projects/my-monorepo/node_modules/broccoli-merge-trees/index.js:16:10)
    at Function.BroccoliMergeTrees [as _upstreamMergeTrees] (/Users/me/Projects/my-monorepo/node_modules/broccoli-merge-trees/index.js:10:53)
    at getEmptyTree (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/broccoli/merge-trees.js:17:37)
    at mergeTrees (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/broccoli/merge-trees.js:82:14)
    at Class.treeFor (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/models/addon.js:638:30)
    at /Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/models/addon.js:497:26
    at Array.reduce (<anonymous>)
    at Class.eachAddonInvoke (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/models/addon.js:494:24)
    at Class.treeFor (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/models/addon.js:627:22)
    at /Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:697:26
    at Array.reduce (<anonymous>)
    at EmberAddon._addonTreesFor (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:695:32)
    at EmberAddon.addonTreesFor (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:719:17)
    at EmberAddon.getAddonTemplates (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:1067:27)
    at EmberAddon.toArray (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:1537:12)
    at EmberAddon.toTree (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:1644:36)
    at EmberAddon.appInstance.toTree (/Users/me/Projects/my-monorepo/node_modules/@embroider/macros/src/ember-addon-main.js:48:35)
    at module.exports (/Users/me/Projects/my-monorepo/my-addon-workspace/ember-cli-build.js:12:14)
    at Builder.readBuildFile (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/models/builder.js:49:14)
    at Builder.setupBroccoliBuilder (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/models/builder.js:63:22)
    at new Builder (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/models/builder.js:29:10)
-~- (end) -~-
  - name: Error
  - nodeAnnotation: EMPTY_MERGE_TREE
  - nodeName: BroccoliMergeTrees
  - originalErrorMessage: [BroccoliMergeTrees: EMPTY_MERGE_TREE] error while merging the following:
ENOENT: no such file or directory, mkdir '/var/folders/jc/pckth1zj4pn8_8_2__qydnc80000gn/T/broccoli-98078d7QG7SJvr6mU/out-000-broccoli_merge_trees_empty_merge_tree'
  - stack: Error: [BroccoliMergeTrees: EMPTY_MERGE_TREE] error while merging the following:
ENOENT: no such file or directory, mkdir '/var/folders/jc/pckth1zj4pn8_8_2__qydnc80000gn/T/broccoli-98078d7QG7SJvr6mU/out-000-broccoli_merge_trees_empty_merge_tree'
    at Object.mkdirSync (fs.js:987:3)
    at update (/Users/me/Projects/my-monorepo/node_modules/fs-updater/index.js:186:10)
    at FSUpdater._update (/Users/me/Projects/my-monorepo/node_modules/fs-updater/index.js:116:7)
    at FSUpdater._update (/Users/me/Projects/my-monorepo/node_modules/fs-updater/index.js:121:12)
    at FSUpdater.update (/Users/me/Projects/my-monorepo/node_modules/fs-updater/index.js:105:10)
    at MergeTrees.merge (/Users/me/Projects/my-monorepo/node_modules/broccoli-merge-trees/node_modules/merge-trees/index.js:31:18)
    at BroccoliMergeTrees.build (/Users/me/Projects/my-monorepo/node_modules/broccoli-merge-trees/index.js:36:21)
    at TransformNodeWrapper.build (/Users/me/Projects/my-monorepo/node_modules/broccoli/dist/wrappers/transform-node.js:71:39)
    at /Users/me/Projects/my-monorepo/node_modules/broccoli/dist/builder.js:185:30
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Builder.build (/Users/me/Projects/my-monorepo/node_modules/broccoli/dist/builder.js:204:13)
    at async Builder.build (/Users/me/Projects/my-monorepo/node_modules/ember-cli/lib/models/builder.js:197:11)
=================================================================================
```

</details>

**Comment by NullVoxPopuli** [comment]
This error occurred while I was converting a single-file-component (using setComponentTemplate) to a "nested" component:

<details><summary>console output</summary>


```
file added components/demo/container
file deleted components/demo/container.ts

Build successful (9139ms) – Serving on https://localhost:6015/

Slowest Nodes (totalTime >= 5%)                                           | Total (avg)       
--------------------------------------------------------------------------+-------------------
Funnel (354)                                                              | 2781ms (7 ms)     
PostcssCompiler (5)                                                       | 832ms (166 ms)    
Package /assets/vendor.js (1)                                             | 782ms             
TreeMerger (addon) (1)                                                    | 483ms             


Build successful (9811ms) – Serving on https://localhost:6015/

Slowest Nodes (totalTime >= 5%)                                           | Total (avg)       
--------------------------------------------------------------------------+-------------------
Funnel (354)                                                              | 2831ms (7 ms)     
PostcssCompiler (5)                                                       | 910ms (182 ms)    
Package /assets/vendor.js (1)                                             | 841ms             

file added components/demo/container/index.hbs
Build Error (ColocatedTemplateProcessor)

ENOENT: no such file or directory, open '/tmp/broccoli-376267rNAt2Hm9ZWk/out-1655-colocated_template_processor/@crowdstrike/styleguide-support/components/demo/container/index.ts'


Stack Trace and Error Report: /tmp/error.dump.94252c697d504fe018db6e3ec98a6fdf.log
internal/process/promises.js:213
        triggerUncaughtException(err, true /* fromPromise */);
        ^

BuildError: ENOENT: no such file or directory, open '/tmp/broccoli-376267rNAt2Hm9ZWk/out-1655-colocated_template_processor/@crowdstrike/styleguide-support/components/demo/container/index.ts'
    at Object.openSync (fs.js:476:3)
    at Object.writeFileSync (fs.js:1467:35)
    at ColocatedTemplateProcessor.build (/my-monorepo/node_modules/ember-cli-htmlbars/lib/colocated-broccoli-plugin.js:194:14)
    at TransformNodeWrapper.build (/my-monorepo/node_modules/broccoli/dist/wrappers/transform-node.js:71:39)
    at /my-monorepo/node_modules/broccoli/dist/builder.js:185:30
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Builder.build (/my-monorepo/node_modules/broccoli/dist/builder.js:204:13)
    at async Builder.build (/my-monorepo/node_modules/ember-cli/lib/models/builder.js:197:11) {
  isBuilderError: true,
  isSilent: undefined,
  isCancellation: undefined,
  broccoliPayload: {
    originalError: Error: ENOENT: no such file or directory, open '/tmp/broccoli-376267rNAt2Hm9ZWk/out-1655-colocated_template_processor/@crowdstrike/styleguide-support/components/demo/container/index.ts'
        at Object.openSync (fs.js:476:3)
        at Object.writeFileSync (fs.js:1467:35)
        at ColocatedTemplateProcessor.build (/my-monorepo/node_modules/ember-cli-htmlbars/lib/colocated-broccoli-plugin.js:194:14)
        at TransformNodeWrapper.build (/my-monorepo/node_modules/broccoli/dist/wrappers/transform-node.js:71:39)
        at /my-monorepo/node_modules/broccoli/dist/builder.js:185:30
        at runMicrotasks (<anonymous>)
        at processTicksAndRejections (internal/process/task_queues.js:93:5)
        at async Builder.build (/my-monorepo/node_modules/broccoli/dist/builder.js:204:13)
        at async Builder.build (/my-monorepo/node_modules/ember-cli/lib/models/builder.js:197:11) {
      errno: -2,
      syscall: 'open',
      code: 'ENOENT',
      path: '/tmp/broccoli-376267rNAt2Hm9ZWk/out-1655-colocated_template_processor/@crowdstrike/styleguide-support/components/demo/container/index.ts'
    },
    originalMessage: "ENOENT: no such file or directory, open '/tmp/broccoli-376267rNAt2Hm9ZWk/out-1655-colocated_template_processor/@crowdstrike/styleguide-support/components/demo/container/index.ts'",
    nodeId: 1655,
    nodeLabel: 'ColocatedTemplateProcessor',
    nodeName: 'ColocatedTemplateProcessor',
    nodeAnnotation: undefined,
    instantiationStack: '    at new Plugin (/my-monorepo/node_modules/ember-cli-htmlbars/node_modules/broccoli-plugin/dist/index.js:45:33)\n' +
      '    at new ColocatedTemplateProcessor (/my-monorepo/node_modules/ember-cli-htmlbars/lib/colocated-broccoli-plugin.js:12:5)\n' +
      '    at Object.toTree (/my-monorepo/node_modules/ember-cli-htmlbars/lib/ember-addon-main.js:94:33)\n' +
      '    at /my-monorepo/node_modules/ember-cli-preprocess-registry/preprocessors.js:188:26\n' +
      '    at Array.forEach (<anonymous>)\n' +
      '    at processPlugins (/my-monorepo/node_modules/ember-cli-preprocess-registry/preprocessors.js:186:11)\n' +
      '    at module.exports.preprocessTemplates (/my-monorepo/node_modules/ember-cli-preprocess-registry/preprocessors.js:169:10)\n' +
      '    at Class.compileTemplates (/my-monorepo/node_modules/ember-cli/lib/models/addon.js:1138:35)\n' +
      '    at Class.compileAddon (/my-monorepo/node_modules/ember-cli/lib/models/addon.js:1200:42)\n' +
      '    at Class.treeForAddon (/my-monorepo/node_modules/ember-cli/lib/models/addon.js:856:26)\n' +
      '    at Class._treeFor (/my-monorepo/node_modules/ember-cli/lib/models/addon.js:668:33)\n' +
      '    at Class.treeFor (/my-monorepo/node_modules/ember-cli/lib/models/addon.js:628:21)\n' +
      '    at /my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:697:26\n' +
      '    at Array.reduce (<anonymous>)\n' +
      '    at EmberApp._addonTreesFor (/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:695:32)\n' +
      '    at EmberApp._addonBundles (/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:1109:31)\n' +
      '    at EmberApp.createAddonTree (/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:1122:29)\n' +
      '    at EmberApp.addonTree (/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:1140:36)\n' +
      '    at EmberApp.getExternalTree (/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:1188:25)\n' +
      '    at EmberApp.toArray (/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:1540:12)\n' +
      '    at EmberApp.toTree (/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:1644:36)\n' +
      '    at module.exports (/my-monorepo/my-app/ember-cli-build.js:45:14)',
    location: {
      file: undefined,
      treeDir: undefined,
      line: undefined,
      column: undefined
    },
    error: {
      message: "ENOENT: no such file or directory, open '/tmp/broccoli-376267rNAt2Hm9ZWk/out-1655-colocated_template_processor/@crowdstrike/styleguide-support/components/demo/container/index.ts'",
      stack: "Error: ENOENT: no such file or directory, open '/tmp/broccoli-376267rNAt2Hm9ZWk/out-1655-colocated_template_processor/@crowdstrike/styleguide-support/components/demo/container/index.ts'\n" +
        '    at Object.openSync (fs.js:476:3)\n' +
        '    at Object.writeFileSync (fs.js:1467:35)\n' +
        '    at ColocatedTemplateProcessor.build (/my-monorepo/node_modules/ember-cli-htmlbars/lib/colocated-broccoli-plugin.js:194:14)\n' +
        '    at TransformNodeWrapper.build (/my-monorepo/node_modules/broccoli/dist/wrappers/transform-node.js:71:39)\n' +
        '    at /my-monorepo/node_modules/broccoli/dist/builder.js:185:30\n' +
        '    at runMicrotasks (<anonymous>)\n' +
        '    at processTicksAndRejections (internal/process/task_queues.js:93:5)\n' +
        '    at async Builder.build (/my-monorepo/node_modules/broccoli/dist/builder.js:204:13)\n' +
        '    at async Builder.build (/my-monorepo/node_modules/ember-cli/lib/models/builder.js:197:11)',
      errorType: 'Build Error',
      codeFrame: "ENOENT: no such file or directory, open '/tmp/broccoli-376267rNAt2Hm9ZWk/out-1655-colocated_template_processor/@crowdstrike/styleguide-support/components/demo/container/index.ts'",
      location: {
        file: undefined,
        treeDir: undefined,
        line: undefined,
        column: undefined
      }
    },
    broccoliNode: {
      nodeName: 'ColocatedTemplateProcessor',
      nodeAnnotation: undefined,
      instantiationStack: '    at new Plugin (/my-monorepo/node_modules/ember-cli-htmlbars/node_modules/broccoli-plugin/dist/index.js:45:33)\n' +
        '    at new ColocatedTemplateProcessor (/my-monorepo/node_modules/ember-cli-htmlbars/lib/colocated-broccoli-plugin.js:12:5)\n' +
        '    at Object.toTree (/my-monorepo/node_modules/ember-cli-htmlbars/lib/ember-addon-main.js:94:33)\n' +
        '    at /my-monorepo/node_modules/ember-cli-preprocess-registry/preprocessors.js:188:26\n' +
        '    at Array.forEach (<anonymous>)\n' +
        '    at processPlugins (/my-monorepo/node_modules/ember-cli-preprocess-registry/preprocessors.js:186:11)\n' +
        '    at module.exports.preprocessTemplates (/my-monorepo/node_modules/ember-cli-preprocess-registry/preprocessors.js:169:10)\n' +
        '    at Class.compileTemplates (/my-monorepo/node_modules/ember-cli/lib/models/addon.js:1138:35)\n' +
        '    at Class.compileAddon (/my-monorepo/node_modules/ember-cli/lib/models/addon.js:1200:42)\n' +
        '    at Class.treeForAddon (/my-monorepo/node_modules/ember-cli/lib/models/addon.js:856:26)\n' +
        '    at Class._treeFor (/my-monorepo/node_modules/ember-cli/lib/models/addon.js:668:33)\n' +
        '    at Class.treeFor (/my-monorepo/node_modules/ember-cli/lib/models/addon.js:628:21)\n' +
        '    at /my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:697:26\n' +
        '    at Array.reduce (<anonymous>)\n' +
        '    at EmberApp._addonTreesFor (/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:695:32)\n' +
        '    at EmberApp._addonBundles (/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:1109:31)\n' +
        '    at EmberApp.createAddonTree (/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:1122:29)\n' +
        '    at EmberApp.addonTree (/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:1140:36)\n' +
        '    at EmberApp.getExternalTree (/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:1188:25)\n' +
        '    at EmberApp.toArray (/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:1540:12)\n' +
        '    at EmberApp.toTree (/my-monorepo/node_modules/ember-cli/lib/broccoli/ember-app.js:1644:36)\n' +
        '    at module.exports (/my-monorepo/my-app/ember-cli-build.js:45:14)'
    },
    versions: { 'broccoli-builder': '3.5.1', node: 'v14.15.4' }
  }
}


```

</details>

<details><summary>ENV Summary in the log with the same info as above</summary>

```
ENV Summary:

  TIME: Wed May 19 2021 13:44:25 GMT-0400 (Eastern Daylight Time)
  TITLE: ember
  ARGV:
  - /home/me/.volta/tools/image/node/14.15.4/bin/node
  - /my-monorepo/my-app/node_modules/.bin/ember
  - serve
  - --watcher
  - polling
  EXEC_PATH: /home/me/.volta/tools/image/node/14.15.4/bin/node
  TMPDIR: /tmp
  SHELL: /bin/bash
  PATH:
  - /home/me/.volta/tools/image/yarn/1.22.5/bin
  - /home/me/.volta/tools/image/node/14.15.4/bin
  - /home/me/.volta/bin
  - /home/me/.pyenv/plugins/pyenv-virtualenv/shims
  - /home/me/.pyenv/shims
  - /home/me/.pyenv/bin
  - /home/me/Development/Work/scripts/
  - /home/me/Applications
  - /home/me/apps/phantomjs/bin
  - /home/me/scripts/system-utils
  - /home/me/scripts/git
  - /home/me/scripts/rails
  - /home/me/scripts
  - /home/me/.volta/bin
  - /home/me/.local/bin
  - /usr/local/sbin
  - /usr/local/bin
  - /usr/sbin
  - /usr/bin
  - /sbin
  - /bin
  - /usr/games
  - /usr/local/games
  - /snap/bin
  - /home/me/.fzf/bin
  - /usr/lib/go-1.13/bin
  - /bin
  - /home/me/go/bin
  PLATFORM: linux x64
  FREEMEM: 4367646720
  TOTALMEM: 16790663168
  UPTIME: 16857
  LOADAVG: 0.73,0.76,0.83
  CPUS:
  - Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz - 2400
  - Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz - 2400
  - Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz - 2400
  - Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz - 2400
  - Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz - 2400
  - Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz - 2400
  - Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz - 2400
  - Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz - 2400
  ENDIANNESS: LE
  VERSIONS:
  - ares: 1.16.1
  - brotli: 1.0.9
  - cldr: 37.0
  - icu: 67.1
  - llhttp: 2.1.3
  - modules: 83
  - napi: 7
  - nghttp2: 1.41.0
  - node: 14.15.4
  - openssl: 1.1.1i
  - tz: 2020a
  - unicode: 13.0
  - uv: 1.40.0
  - v8: 8.4.371.19-node.17
  - zlib: 1.2.11

```

</details>

**Comment by NullVoxPopuli** [comment]
Closing as duplicate of: https://github.com/ember-cli/ember-cli/issues/9404
